src/cosimulation.cc

```diff
@@ -23,8 +55,12 @@ ComplexCoDRAMsim3::~ComplexCoDRAMsim3() {
 }
 
 void ComplexCoDRAMsim3::tick() {
-    memory->ClockTick();
-    dram_clock++;
+    cpu_time += CPU_CLOCK_CYCLE;
+    if (cpu_time >= next_dram_cycle_time) {
+        next_dram_cycle_time += DRAM_CLOCK_CYCLE;
+        memory->ClockTick();
+        dram_clock++;
```

Comment:
I think this `dram_clock` is actually CPU clock (cycle). `dram_clock` was used to track the padding cycles in the CPU clock. It is more like the `cpu_time` but for counting the cycles instead of the time. So probably the dram_clock here is incorrectly implemented.

Reply:
You are right:
`resp->finish_time = resp->req_time + (get_clock_ticks() - resp->req_time) + padding;`